open-apparel-registry/open-apparel-registry
# Validate claimed facility website field and show as hyperlink

## Overview

Adds client validation of the facility claim website field, changes the related model field to a URLField, and makes valid URLs clickable on the facility details sidebar.

Connects #587 

## Demo

### Form validation

<img width="838" alt="Screen Shot 2019-07-02 at 4 56 40 PM" src="https://user-images.githubusercontent.com/17363/60553966-8c33dc00-9cea-11e9-82b1-7ecf38b72cef.png">

### Clickable link

<img width="530" alt="Screen Shot 2019-07-02 at 4 56 17 PM" src="https://user-images.githubusercontent.com/17363/60553971-90f89000-9cea-11e9-84b8-e98fb39035e1.png">

## Testing Instructions

Assumes that the fixtures have been loaded and processed.

* Run `./scripts/update`
* Make sure facility claiming is enabled
  * `./scripts/manage waffle_switch claim_a_facility on`
  * Restart `./scripts/server`
* Log in as `c2@example.com`
* Claim a facility
  * On the "Facility Information" step, verify that you can only proceed to the next step if the website field is either empty or a valid url (with or without a scheme)
* Log in as `c1@example.com` and approve the claim via http://localhost:6543/dashboard/claims
* Log back in as `c2@example.com` and open the facility claim details page via http://localhost:6543/claimed
* Verify that you can only enter valid URLs into the website field (protocol is optional)
  * Verify that the save button is disabled until the website field is valid
* Enter a valid URL in the website field and save the form.
* Browse the facility details page and verify that the website is a clickable link that opens the URL in a new tab/window.

## Checklist
